resolvers.py

```diff
@@ -0,0 +1,1979 @@
+            leave_type="loss of pay", leave_end_date__gte=cw_start, leave_start_date__lte=cw_end
+        )
+        qs_filters_maternity = Q(
+            leave_type="maternity",
+            leave_end_date__gte=cw_start,
+            leave_start_date__lte=cw_end,
+        )
+
+        if column_name.value == "RESIGNED":
+            qs_filters = qs_filters_resigned
+
+        if column_name.value == "LOP":
+            qs_filters = qs_filters_lop
+
+        if column_name.value == "MATERNITY":
+            qs_filters = qs_filters_maternity
+
+        if column_name.value == "NON_DEPLOYABLE_TOTAL":
+            qs_filters = qs_filters_resigned | qs_filters_lop | qs_filters_maternity
+
+        qs = models.BaseBenchDataSource.objects.filter(qs_filters)
+        qs = qs.filter(band__in=BENCH_REPORTS_BAND,
+                       on_bench__in=ON_BENCH, allocation_end_dates__isnull=True)
+        qs = qs.filter(department='Delivery')
+
+        if str(band_name).lower() == 'total':
+            return qs
+
+        else:
+            qs = qs.filter(band=band_name)
+
+        return qs
+
+
+# Cell Details Resolver for column names "Allocation Pending", "Nominated", "Bench", "Deployable Total (Grand Total)"
+# for Overall Bench View By Capability
+
+class BaseDataSourceResolverForOverallBenchByCapabilityDeployable:
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        # params required
+        # 1. capability name
+        # 2. column (Allocation Pending, Nominated, Bench, Deployable_Total)
+
+        capability_name = kwargs.get("capability_name")
+        column_name = kwargs.get("column_name")
+
+        qs_filters = Q(nomination_status__in=column_name.value)
+
+        if None in column_name.value:
+            qs_filters = qs_filters | Q(nomination_status__isnull=True)
+
+        qs = models.BaseBenchDataSource.objects.filter(qs_filters)
+        qs = qs.filter(updated_end_date__isnull=False)
+        qs = qs.filter(band__in=BENCH_REPORTS_BAND,
+                       on_bench__in=ON_BENCH, allocation_end_dates__isnull=True)
+        qs = qs.filter(department='Delivery')
+
+        if str(capability_name).lower() == 'other':
+            # handle case like "Java,QA"
+
+            return qs.filter(capabilities=None)
+        elif str(capability_name).lower() == "total":
+            return qs
+
+        return qs.filter(capabilities=capability_name)
+
+
+# Cell Details Resolver for column names "Resigned", "LOP", "Maternity", "Non Deployable Total (Grand Total)"
+# for Overall Bench View By Capability
+
+class BaseDataSourceResolverForOverallBenchByCapabilityNonDeployable:
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        # params required
+        # 1. capability name
+        # 2. column (Resigned, loss of pay, maternity)
+
+        capability_name = kwargs.get("capability_name")
+        column_name = kwargs.get("column_name")
+        today = date.today()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Naive datetime objects are treated by many datetime methods as local times, We recommend that you create aware timestamps with the `now` or `fromtimestamp` functions using the `tzinfo` to attribute to represent times in UTC. Do not use the [datetime.utcnow()](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow) because it returns only a `naive` datetime object. 

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)